Compilation fails on Node 12

https://travis-ci.org/ioBroker/ioBroker.zwave/jobs/527514030#L686

```
make: Entering directory `/home/travis/build/ioBroker/ioBroker.zwave/node_modules/openzwave-shared/build'
  CXX(target) Release/obj.target/openzwave_shared/src/callbacks.o
  CXX(target) Release/obj.target/openzwave_shared/src/openzwave.o
../src/openzwave.cc:34:23: error: variable or field ‘init’ declared void
  extern "C" void init(Handle<Object> target, Handle<Object> module) {
                       ^
../src/openzwave.cc:34:23: error: ‘Handle’ was not declared in this scope
../src/openzwave.cc:34:36: error: expected primary-expression before ‘>’ token
  extern "C" void init(Handle<Object> target, Handle<Object> module) {
                                    ^
../src/openzwave.cc:34:38: error: ‘target’ was not declared in this scope
  extern "C" void init(Handle<Object> target, Handle<Object> module) {
                                      ^
../src/openzwave.cc:34:46: error: ‘Handle’ was not declared in this scope
  extern "C" void init(Handle<Object> target, Handle<Object> module) {
                                              ^
../src/openzwave.cc:34:59: error: expected primary-expression before ‘>’ token
  extern "C" void init(Handle<Object> target, Handle<Object> module) {
                                                           ^
../src/openzwave.cc:34:61: error: ‘module’ was not declared in this scope
  extern "C" void init(Handle<Object> target, Handle<Object> module) {
                                                             ^
In file included from /home/travis/build/ioBroker/ioBroker.zwave/node_modules/nan/nan.h:53:0,
                 from ../src/openzwave.hpp:27,
                 from ../src/openzwave.cc:18:
/home/travis/.node-gyp/12.1.0/include/node/node.h:565:3: error: expected ‘}’ at end of input
   }
   ^
/home/travis/.node-gyp/12.1.0/include/node/node.h:590:3: note: in expansion of macro ‘NODE_MODULE_X’
   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
   ^
../src/openzwave.cc:266:1: note: in expansion of macro ‘NODE_MODULE’
 NODE_MODULE(openzwave_shared, OZW::init)
 ^
make: *** [Release/obj.target/openzwave_shared/src/openzwave.o] Error 1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -34,8 +34,8 @@ Fix scene values type 'list' (Partially fix #266) #267
 add support for SmartOS #272
 
 # 1.6.0
-Fix broken Node 12 builds (v8 7.4.xxx)
-Initial support for OpenZWave 1.6
+Fix broken Node 12 builds (v8 7.4.xxx) #302
+Initial support for OpenZWave 1.6 #300
   - implement exception macros
   - [deprecate several legacy OpenZWave calls](https://github.com/OpenZWave/open-zwave/wiki/OpenZWave-1.6-Release-Notes#deprecated-featuresmethods)
 Updated Raspbian Readme to fix bug when installing on Raspbian stretch #271

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -34,6 +34,7 @@ Fix scene values type 'list' (Partially fix #266) #267
 add support for SmartOS #272
 
 # 1.6.0
+Fix broken Node 12 builds (v8 7.4.xxx)
 Initial support for OpenZWave 1.6
   - implement exception macros
   - [deprecate several legacy OpenZWave calls](https://github.com/OpenZWave/open-zwave/wiki/OpenZWave-1.6-Release-Notes#deprecated-featuresmethods)

--- a/binding.gyp
+++ b/binding.gyp
@@ -56,7 +56,9 @@
 					"OPENZWAVE_DOC=<!@(node -p -e \"'<(OZW_DOC)'.length ? '<(OZW_DOC)' : '/usr/local/share/doc/openzwave'\")",
 					"OPENZWAVE_SECURITY=<!@(find <(OZW_INC) -name ZWSecurity.h | wc -l)",
 					"OPENZWAVE_EXCEPTIONS=<!@(find <(OZW_INC) -name OZWException.h | wc -l)",
-					"OPENZWAVE_BITSET=<!@(find <(OZW_INC) -name ValueBitSet.h | wc -l)"
+					"OPENZWAVE_BITSET=<!@(find <(OZW_INC) -name ValueBitSet.h | wc -l)",
+					"OPENZWAVE_VALUETYPE_FROM_ENUM=<!@(ldconfig -p | grep 'libopenzwave.so ' | awk '{print $4}' | xargs nm -g --demangle |grep OpenZWave::Value::GetTypeNameFromEnum | wc -l)",
+					"OPENZWAVE_VALUETYPE_FROM_VALUEID=<!@(ldconfig -p | grep 'libopenzwave.so ' | awk '{print $4}' | xargs nm -g --demangle |grep OpenZWave::ValueID::GetTypeAsString | wc -l)",
 				],
 				"link_settings": {
 				    "libraries": [
@@ -89,6 +91,8 @@
 					"OPENZWAVE_SECURITY=<!@(find <(OZW_INC) -name ZWSecurity.h | wc -l)",
 					"OPENZWAVE_EXCEPTIONS=<!@(find <(OZW_INC) -name OZWException.h | wc -l)",
 					"OPENZWAVE_BITSET=<!@(find <(OZW_INC) -name ValueBitSet.h | wc -l)",
+					"OPENZWAVE_VALUETYPE_FROM_ENUM=<!@(ldconfig -p | grep 'libopenzwave.so ' | awk '{print $4}' | xargs nm -g --demangle |grep OpenZWave::Value::GetTypeNameFromEnum | wc -l)",
+					"OPENZWAVE_VALUETYPE_FROM_VALUEID=<!@(ldconfig -p | grep 'libopenzwave.so ' | awk '{print $4}' | xargs nm -g --demangle |grep OpenZWave::ValueID::GetTypeAsString | wc -l)",
 				],
 				"link_settings": {
 					"libraries": ["-lopenzwave"]
@@ -121,7 +125,8 @@
 					"OPENZWAVE_ETC=<(OZW_HOME)/config",
 					"OPENZWAVE_SECURITY=1",
 					"OPENZWAVE_EXCEPTIONS=1",
-					"OPENZWAVE_BITSET=1"
+					"OPENZWAVE_BITSET=1",
+					"OPENZWAVE_VALUETYPE_FROM_VALUEID=1"
 				],
 				'msvs_settings': {
 					'VCCLCompilerTool': {

--- a/src/openzwave-config.cc
+++ b/src/openzwave-config.cc
@@ -29,7 +29,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(3, "nodeid, param, value");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		uint8 param  = Nan::To<Number>(info[1]).ToLocalChecked()->Value();
 		int32 value  = Nan::To<Number>(info[2]).ToLocalChecked()->Value();
 		if (info.Length() < 4) {

--- a/src/openzwave-driver.cc
+++ b/src/openzwave-driver.cc
@@ -77,10 +77,12 @@ namespace OZW {
 			Local < Object > opts = Nan::To<Object>(info[0]).ToLocalChecked();
 			Local < Array > props = Nan::GetOwnPropertyNames(opts).ToLocalChecked();
 			for (unsigned int i = 0; i < props->Length(); ++i) {
-				Local<Value> key       = props->Get(i);
-				::std::string  keyname   = *Nan::Utf8String(key);
-				Local<Value> argval    = Nan::Get(opts, key).ToLocalChecked();
-				::std::string  argvalstr = *Nan::Utf8String(argval);
+				Nan::MaybeLocal<Value> keymaybe =  Nan::Get(props, i);
+				if(keymaybe.IsEmpty()) continue;
+				Local<Value>  key       = keymaybe.ToLocalChecked();
+				::std::string keyname   = *Nan::Utf8String(key);
+				Local<Value>  argval    = Nan::Get(opts, key).ToLocalChecked();
+				::std::string argvalstr = *Nan::Utf8String(argval);
 				// UserPath is directly passed to Manager->Connect()
 				// scan for OpenZWave options.xml in the nodeJS module's '/config' subdirectory
 				if (keyname == "UserPath") {
@@ -88,7 +90,7 @@ namespace OZW {
 				} else if (keyname == "ConfigPath") {
 					ozw_config_path.assign(argvalstr);
 				} else if (keyname == "LogInitialisation") {
-					log_initialisation = argval->BooleanValue();
+					log_initialisation = (Nan::To<bool>(argval) == Nan::Just(true));
 				} else {
 					option_overrides += " --" + keyname + " " + argvalstr;
 				}

--- a/src/openzwave-groups.cc
+++ b/src/openzwave-groups.cc
@@ -33,7 +33,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		uint8 numGroups = 0;
 		OZWManagerAssign(numGroups, GetNumGroups, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Integer>(numGroups));
@@ -49,8 +49,8 @@ namespace OZW {
 		Nan::HandleScope scope;
 		CheckMinArgs(2, "nodeid, groupidx");
 		uint8* associations;
-		uint8 nodeid = info[0]->Uint32Value();
-		uint8 groupidx = info[1]->Uint32Value();
+		uint8 nodeid   = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
+		uint8 groupidx = Nan::To<Number>(info[1]).ToLocalChecked()->Value();
 
 		uint32 numNodes = 0;
 		OZWManagerAssign(numNodes, GetAssociations,
@@ -60,7 +60,7 @@ namespace OZW {
 		Local<Array> o_assocs = Nan::New<Array>(numNodes);
 
 		for (uint8 nr = 0; nr < numNodes; nr++) {
-			o_assocs->Set(Nan::New<Integer>(nr), Nan::New<Integer>(associations[nr]));
+			Nan::Set(o_assocs, nr, Nan::New<Integer>(associations[nr]));
 		}
 		if (numNodes > 0) {
 			// The caller is responsible for freeing the array memory with a call to delete [].
@@ -79,8 +79,8 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(2, "nodeid, groupidx");
-		uint8 nodeid = info[0]->Uint32Value();
-		uint8 groupidx = info[1]->Uint32Value();
+		uint8 nodeid   = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
+		uint8 groupidx = Nan::To<Number>(info[1]).ToLocalChecked()->Value();
 
 		uint8 numMaxAssoc = 0;
 		OZWManagerAssign(numMaxAssoc, GetMaxAssociations,
@@ -99,8 +99,8 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(2, "nodeid, groupidx");
-		uint8 nodeid = info[0]->Uint32Value();
-		uint8 groupidx = info[1]->Uint32Value();
+		uint8 nodeid   = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
+		uint8 groupidx = Nan::To<Number>(info[1]).ToLocalChecked()->Value();
 
 		::std::string groupLabel("");
 		OZWManagerAssign(groupLabel, GetGroupLabel,
@@ -122,12 +122,12 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(3, "nodeid, groupidx, tgtnodeid");
-		uint8 nodeid = info[0]->Uint32Value();
-		uint8 groupidx = info[1]->Uint32Value();
-		uint8 tgtnodeid = info[2]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
+		uint8 groupidx = Nan::To<Number>(info[1]).ToLocalChecked()->Value();
+		uint8 tgtnodeid = Nan::To<Number>(info[2]).ToLocalChecked()->Value();
 		uint8 instanceid = 0;
 		if(info.Length() > 3) {
-			instanceid = info[3]->Uint32Value();
+			instanceid = Nan::To<Number>(info[3]).ToLocalChecked()->Value();
 		} 
 
 		OZWManager( AddAssociation,
@@ -144,12 +144,12 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(3, "nodeid, groupidx, tgtnodeid");
-		uint8 nodeid = info[0]->Uint32Value();
-		uint8 groupidx = info[1]->Uint32Value();
-		uint8 tgtnodeid = info[2]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
+		uint8 groupidx = Nan::To<Number>(info[1]).ToLocalChecked()->Value();
+		uint8 tgtnodeid = Nan::To<Number>(info[2]).ToLocalChecked()->Value();
 		uint8 instanceid = 0;
 		if(info.Length() > 3) {
-			instanceid = info[3]->Uint32Value();
+			instanceid = Nan::To<Number>(info[3]).ToLocalChecked()->Value();
 		} 
 
 		OZWManager( RemoveAssociation,

--- a/src/openzwave-network.cc
+++ b/src/openzwave-network.cc
@@ -31,8 +31,9 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
-		uint8 nummsg = (info.Length() > 1) ? info[1]->Uint32Value() : 1;
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
+		uint8 nummsg = (info.Length() > 1) ? 
+			Nan::To<Number>(info[1]).ToLocalChecked()->Value() : 1;
 		OZWManager( TestNetworkNode, homeid, nodeid, nummsg);
 	}
 
@@ -45,7 +46,8 @@ namespace OZW {
 	// ===================================================================
 	{
 		Nan::HandleScope scope;
-		uint8 nummsg = (info.Length() > 0) ? info[0]->Uint32Value() : 1;
+		uint8 nummsg = (info.Length() > 0) ? 
+			Nan::To<Number>(info[0]).ToLocalChecked()->Value() : 1;
 		OZWManager( TestNetwork, homeid, nummsg);
 	}
 
@@ -58,8 +60,9 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
-		uint8 doRR = (info.Length() > 1) ? Nan::To<Boolean>(info[1]).ToLocalChecked()->Value() : false;
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
+		uint8 doRR = (info.Length() > 1) ? 
+			Nan::To<Boolean>(info[1]).ToLocalChecked()->Value() : false;
 		OZWManager( HealNetworkNode, homeid, nodeid, doRR);
 	}
 
@@ -73,7 +76,8 @@ namespace OZW {
 	// ===================================================================
 	{
 		Nan::HandleScope scope;
-		bool doRR = (info.Length() > 0) ? Nan::To<Boolean>(info[0]).ToLocalChecked()->Value() : false;
+		bool doRR = (info.Length() > 0) ? 
+			Nan::To<Boolean>(info[0]).ToLocalChecked()->Value() : false;
 		OZWManager( HealNetwork, homeid, doRR);
 	}
 }

--- a/src/openzwave-nodes.cc
+++ b/src/openzwave-nodes.cc
@@ -31,13 +31,13 @@ namespace OZW {
 		Nan::HandleScope scope;
 		uint8* neighbors;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		uint8 numNeighbors = -1;
 		OZWManagerAssign(numNeighbors, GetNodeNeighbors, homeid, nodeid, &neighbors);
 		Local<Array> o_neighbors = Nan::New<Array>(numNeighbors);
 		if (numNeighbors > 0) {
 			for (uint8 nr = 0; nr < numNeighbors; nr++) {
-				o_neighbors->Set(Nan::New<Integer>(nr), Nan::New<Integer>(neighbors[nr]));
+				Nan::Set(o_neighbors, nr, Nan::New<Integer>(neighbors[nr]));
 			}
 			delete neighbors;
 		}
@@ -51,7 +51,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		OZWManager( SetNodeOn, homeid, nodeid);
 	}
 
@@ -61,7 +61,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		OZWManager( SetNodeOff, homeid, nodeid);
 	}
 
@@ -124,7 +124,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(2, "nodeid, location");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string location(*Nan::Utf8String( info[1] ));
 		OZWManager( SetNodeLocation, homeid, nodeid, location);
 	}
@@ -138,7 +138,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(2, "nodeid, name");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string name(*Nan::Utf8String( info[1] ));
 		OZWManager( SetNodeName, homeid, nodeid, name);
 	}
@@ -159,7 +159,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		OZWManager( RefreshNodeInfo, homeid, nodeid);
 	}
 
@@ -172,7 +172,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string result("");
 		OZWManagerAssign(result,  GetNodeManufacturerName, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<String>(result.c_str()).ToLocalChecked());
@@ -187,7 +187,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		bool result = false;
 		OZWManagerAssign(result,  RequestNodeState, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Boolean>(result));
@@ -202,7 +202,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		bool result = false;
 		OZWManagerAssign(result,  RequestNodeDynamic, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Boolean>(result));
@@ -217,7 +217,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		bool result = false;
 		OZWManagerAssign(result,  IsNodeListeningDevice, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Boolean>(result));
@@ -232,7 +232,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		bool result = false;
 		OZWManagerAssign(result,  IsNodeFrequentListeningDevice, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Boolean>(result));
@@ -247,7 +247,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		bool result = false;
 		OZWManagerAssign(result,  IsNodeBeamingDevice, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Boolean>(result));
@@ -262,7 +262,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		bool result = false;
 		OZWManagerAssign(result,  IsNodeRoutingDevice, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Boolean>(result));
@@ -277,7 +277,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		bool result = false;
 		OZWManagerAssign(result,  IsNodeSecurityDevice, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Boolean>(result));
@@ -292,7 +292,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8  nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		uint32 result = -1;
 		OZWManagerAssign(result,  GetNodeMaxBaudRate, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Uint32>(result));
@@ -307,7 +307,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		uint8 result = -1;
 		OZWManagerAssign(result,  GetNodeVersion, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Integer>(result));
@@ -322,7 +322,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		uint8 result = -1;
 		OZWManagerAssign(result,  GetNodeSecurity, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Integer>(result));
@@ -337,7 +337,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		uint8 result = -1;
 		OZWManagerAssign(result,  GetNodeBasic, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Integer>(result));
@@ -352,7 +352,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		uint8 result = -1;
 		OZWManagerAssign(result,  GetNodeGeneric, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Integer>(result));
@@ -367,7 +367,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		uint8 result = -1;
 		OZWManagerAssign(result, GetNodeSpecific, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<Integer>(result));
@@ -382,7 +382,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string result("");
 		OZWManagerAssign(result,  GetNodeType, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<String>(result.c_str()).ToLocalChecked());
@@ -397,7 +397,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string result("");
 		OZWManagerAssign(result, GetNodeProductName, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<String>(result.c_str()).ToLocalChecked());
@@ -412,7 +412,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string result("");
 		OZWManagerAssign(result, GetNodeName, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<String>(result.c_str()).ToLocalChecked());
@@ -427,7 +427,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string result("");
 		OZWManagerAssign(result, GetNodeLocation, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<String>(result.c_str()).ToLocalChecked());
@@ -442,7 +442,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string result("");
 		OZWManagerAssign(result, GetNodeManufacturerId, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<String>(result.c_str()).ToLocalChecked());
@@ -457,7 +457,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string result("");
 		OZWManagerAssign(result, GetNodeProductType, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<String>(result.c_str()).ToLocalChecked());
@@ -472,7 +472,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string result("");
 		OZWManagerAssign(result, GetNodeProductId, homeid, nodeid);
 		info.GetReturnValue().Set(Nan::New<String>(result.c_str()).ToLocalChecked());
@@ -487,7 +487,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string name(*Nan::Utf8String( info[1] ));
 		OZWManager( SetNodeManufacturerName, homeid, nodeid, name);
 	}
@@ -501,7 +501,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "nodeid");
-		uint8 nodeid = info[0]->Uint32Value();
+		uint8 nodeid = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		::std::string name(*Nan::Utf8String( info[1] ));
 		OZWManager( SetNodeProductName, homeid, nodeid, name);
 	}

--- a/src/openzwave-polling.cc
+++ b/src/openzwave-polling.cc
@@ -50,7 +50,7 @@ namespace OZW {
 	{
 		Nan::HandleScope scope;
 		CheckMinArgs(1, "intervalMillisecs");
-		uint32 intervalMillisecs = info[0]->Uint32Value();
+		uint32 intervalMillisecs = Nan::To<Number>(info[0]).ToLocalChecked()->Value();
 		OZWManager( SetPollInterval, intervalMillisecs, false);
 	}
 

--- a/src/openzwave-values.cc
+++ b/src/openzwave-values.cc
@@ -45,7 +45,7 @@ namespace OZW {
 					break;
 				}
 				case OpenZWave::ValueID::ValueType_Decimal: {
-					float val = Nan::To<Number>(info[validx]).ToLocalChecked()->NumberValue();
+					float val = Nan::To<Number>(info[validx]).ToLocalChecked()->Value();
 					OZWManager( SetValue, *vit, val);
 					break;
 				}
@@ -237,7 +237,9 @@ namespace OZW {
 				if ((info.Length() < idxpos) || !info[idxpos]->IsObject()) {
 					Nan::ThrowTypeError("must supply a switchpoint object");
 				} else {
-					Local<Object> sp = info[idxpos]->ToObject();
+					Nan::MaybeLocal<v8::Object> sp_maybe = Nan::To<v8::Object>(info[idxpos]);
+					if (sp_maybe.IsEmpty()) return;
+					v8::Local<v8::Object> sp = sp_maybe.ToLocalChecked();
 					OZWManager( SetSwitchPoint, *vit,
 						Nan::To<Number>(Nan::Get(sp, Nan::New<String>("hours").ToLocalChecked()).ToLocalChecked()).ToLocalChecked()->Value(),
 						Nan::To<Number>(Nan::Get(sp, Nan::New<String>("minutes").ToLocalChecked()).ToLocalChecked()).ToLocalChecked()->Value(),

--- a/src/openzwave.cc
+++ b/src/openzwave.cc
@@ -31,7 +31,7 @@ namespace OZW {
 	::std::string ozw_config_path  = stringify( OPENZWAVE_ETC );
 
 	// ===================================================================
-	extern "C" void init(Handle<Object> target, Handle<Object> module) {
+	extern "C" void init(Local<Object> target, Local<Object> module) {
 
 		Nan::HandleScope scope;
 
@@ -187,8 +187,12 @@ namespace OZW {
 		Nan::SetPrototypeMethod(t, "sceneGetValues", OZW::SceneGetValues);
 		Nan::SetPrototypeMethod(t, "activateScene", OZW::ActivateScene);
 #endif
-		//
-		Nan::Set(target, Nan::New<String>("Emitter").ToLocalChecked(), t->GetFunction());
+		// register this class as an emitter
+		Nan::Set(
+			target,
+			Nan::New<String>("Emitter").ToLocalChecked(),
+			Nan::GetFunction(t).ToLocalChecked()
+		);
 		/* for BeginControllerCommand
 	   * http://openzwave.com/dev/classOpenZWave_1_1Manager.html#aa11faf40f19f0cda202d2353a60dbf7b
 	   */
@@ -236,7 +240,9 @@ namespace OZW {
 			Nan::MaybeLocal <v8::Array> propsmaybe =  Nan::GetOwnPropertyNames(opts);
 			Local < Array > props = propsmaybe.ToLocalChecked();
 			for (unsigned int i = 0; i < props->Length(); ++i) {
-				Local<Value>  key       = props->Get(i);
+				Nan::MaybeLocal<Value> keymaybe =  Nan::Get(props, i);
+				if(keymaybe.IsEmpty()) continue;
+				Local<Value>  key       = keymaybe.ToLocalChecked();
 				::std::string keyname   = *Nan::Utf8String(key);
 				Local<Value>  argval    = Nan::Get(opts, key).ToLocalChecked();
 				::std::string argvalstr = *Nan::Utf8String(argval);
@@ -247,7 +253,7 @@ namespace OZW {
 				} else if (keyname == "ConfigPath") {
 					ozw_config_path.assign(argvalstr);
 				} else if (keyname == "LogInitialisation") {
-					self->log_initialisation = argval->BooleanValue();
+					self->log_initialisation = (Nan::To<bool>(argval) == Nan::Just(true));
 				} else {
 					option_overrides += " --" + keyname + " " + argvalstr;
 				}

--- a/src/utils.cc
+++ b/src/utils.cc
@@ -252,8 +252,16 @@ namespace OZW {
 		AddStringProp(nodeobj,  value_id,  buffer.c_str());
 		AddIntegerProp(nodeobj, node_id,   value.GetNodeId());
 		AddIntegerProp(nodeobj, class_id,  value.GetCommandClassId());
+#if OPENZWAVE_VALUETYPE_FROM_VALUEID
+		AddStringProp (nodeobj, type,      value.GetTypeAsString());
+		AddStringProp (nodeobj, genre,     value.GetGenreAsString());
+#elif OPENZWAVE_VALUETYPE_FROM_ENUM
 		AddStringProp (nodeobj, type,      OpenZWave::Value::GetTypeNameFromEnum(value.GetType()));
 		AddStringProp (nodeobj, genre,     OpenZWave::Value::GetGenreNameFromEnum(value.GetGenre()));
+#else
+		AddIntegerProp(nodeobj, type,      value.GetType());
+		AddIntegerProp(nodeobj, genre,     value.GetGenre());
+#endif
 		AddIntegerProp(nodeobj, instance,  value.GetInstance());
 		AddIntegerProp(nodeobj, index,     value.GetIndex());
 		AddStringProp (nodeobj, label,     mgr->GetValueLabel(value).c_str());

--- a/src/utils.hpp
+++ b/src/utils.hpp
@@ -55,7 +55,7 @@
 #define AddBooleanProp(OBJ,PROPNAME,PROPVALUE) \
 	Nan::Set(OBJ,                                \
 		Nan::New<v8::String>( #PROPNAME ).ToLocalChecked(),  \
-		Nan::New<v8::Boolean>( PROPVALUE )->ToBoolean());
+		Nan::New<v8::Boolean>( PROPVALUE ));
 
 #define AddStringProp(OBJ,PROPNAME,PROPVALUE) \
 	Nan::Set(OBJ,                               \

